src/SignalR/clients/ts/signalr/src/HubConnection.ts

```diff
@@ -120,6 +120,17 @@ export class HubConnection {
         return this.connectionState;
     }
 
+    /** Represents the connection id of the {@link HubConnection} on the server. The connection id will be null when the connection is either
+     *  in the disconnected state or if the negotiation step was skipped.
+     */
+    get connectionId(): string | null {
+        if (this.connection.connectionId) {
```

Comment:
Can connection be null?

Reply:
The property comment explains when connectionId could be null.

Reply:
connection not connection property. If connection is null it'll null ref.

Reply:
The underlying connection should always be defined

Reply:
But we can add a check here if you want

Reply:
We don't need it if it's never null